mcproto/decode.go

```diff
@@ -58,9 +60,65 @@ func DecodeLoginStart(data interface{}) (*LoginStart, error) {
 		return loginStart, errors.Wrap(err, "failed to read username")
 	}
 
-	loginStart.PlayerUuid, err = ReadUuid(buffer)
-	if err != nil {
-		return loginStart, errors.Wrap(err, "failed to read player uuid")
+	// These versions can send player keypair data. Ignore it.
+	// Reference: https://github.com/MCCTeam/Minecraft-Console-Client/blob/f785f509f228bf787c237ac139e6f666a960819a/MinecraftClient/Protocol/Handlers/Protocol18.cs#L2808-L2828
+	if protocolVersion >= ProtocolVersion1_19 && protocolVersion <= ProtocolVersion1_19_2 {
+		hasSignatureData, err := ReadBoolean(buffer)
+		if err != nil {
+			return loginStart, errors.Wrap(err, "failed to read has signature data flag")
+		}
+
+		if hasSignatureData {
+			// Read and discard the data
+			_, err = ReadLong(buffer) // Expiration time
+			if err != nil {
+				return loginStart, errors.Wrap(err, "failed to read expiration time")
+			}
+
+			pubKeyLength, err := ReadVarInt(buffer) // Length of the public key
+			if err != nil {
+				return loginStart, errors.Wrap(err, "failed to read public key length")
+			}
+
+			_, err = ReadByteArray(buffer, pubKeyLength) // Public key data
+			if err != nil {
+				return loginStart, errors.Wrap(err, "failed to read public key")
+			}
+
+			signatureLength, err := ReadVarInt(buffer) // Length of the signature
+			if err != nil {
+				return loginStart, errors.Wrap(err, "failed to read signature length")
+			}
+
+			_, err = ReadByteArray(buffer, signatureLength) // Signature data
+			if err != nil {
+				return loginStart, errors.Wrap(err, "failed to read signature")
+			}
+		}
+	}
+
+	// Reference: https://github.com/MCCTeam/Minecraft-Console-Client/blob/f785f509f228bf787c237ac139e6f666a960819a/MinecraftClient/Protocol/Handlers/Protocol18.cs#L2831-L2853
```

Comment:
And 1.19.2

https://minecraft.wiki/w/Minecraft_Wiki:Projects/wiki.vg_merge/Protocol?oldid=2772944#Login_Start